homeassistant/components/unifiled/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for unifiled integration."""
+import logging
+
+from unifiled import unifiled
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str, "port": str, "username": str, "password": str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    try:
+        unifiled(
```

Comment:
This is doing I/O inside async. Use executor.

Reply:
replacing that line with `api = await hass.async_add_executor_job(unifiled(data["host"], data["port"], data["username"], data["password"]))` gives me this: ```
Traceback (most recent call last):
  File "/home/floris/home-assistant/homeassistant/components/unifiled/config_flow.py", line 48, in async_step_user
    info = await validate_input(self.hass, user_input)
  File "/home/floris/home-assistant/homeassistant/components/unifiled/config_flow.py", line 28, in validate_input
    api = await hass.async_add_executor_job(unifiled(data["host"], data["port"], data["username"], data["password"]))
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: 'unifiled' object is not callable
```

Any ideas ?

Reply:
Try:
```py
hass.async_add_executor_job(
    unifiled, data["host"], data["port"], data["username"], data["password"]
)
```

Reply:
That fixed it.